docs/graph_engine/data_spec.rst

```diff
@@ -17,9 +17,77 @@ Users can generate a graph in either format then our pipeline will convert it in
 DeepGNN also supports writing custom decoders, see [the decoders file](https://github.com/microsoft/DeepGNN/blob/main/src/python/deepgnn/graph_engine/snark/decoders.py).
 Just inheret the base class Decoder, overwrite the decode function and pass the new decoder as an argument to the converter or dispatcher.
 
-1. `JSON <#json-format>`_: Heterogeneous or homegeneous graph.
+1. `Linear <#linear-format>`_: Heterogeneous or homegeneous graph.
 
-2. `TSV <#tsv-format>`_: Homogeneous graph only.
+2. `JSON <#json-format>`_: Heterogeneous or homegeneous graph.
+
+3. `TSV <#tsv-format>`_: Homogeneous graph only.
+
+Linear Format
+===========
+
+Linear format is most directly what is given to binary writers, it supports nodes and their outgoing edges in
```

Comment:
Please, avoid using new terminology/implementation detail right away from the start. What is binary writer and why it is important to describe a linear format?